### PR TITLE
:book: Add troubleshooting page to the doc

### LINKED ISSUE #1719: Can't apply node-role.kubernetes.io at bootstrap time
**User Story**

As an operator, I would like to be able to put `node-role.kubernetes.io` label at creation time through `kubeadm`

**Detailed Description**

I would like to be able to give a `role` to my nodes (eg master, worker) at bootstrap. Unfortunately, I got an error with `kubeadm` when I try to apply such label through the config file:

```
---
apiVersion: kubeadm.k8s.io/v1beta2
kind: JoinConfiguration
discovery:
  bootstrapToken:
    apiServerEndpoint: ""
    token: ""
    unsafeSkipCAVerification: true
nodeRegistration:
  kubeletExtraArgs:
    node-labels: node-role.kubernetes.io/worker=""
```

```
Nov  2 06:56:02 dday-instance-2 kubelet[6823]: --node-labels in the 'kubernetes.io' namespace must begin with an allowed prefix (kubelet.kubernetes.io, node.kubernetes.io) or be in the specifically allowed set (beta.kubernetes.io/arch, beta.kubernetes.io/instance-type, beta.kubernetes.io/os, failure-domain.beta.kubernetes.io/region, failure-domain.beta.kubernetes.io/zone, failure-domain.kubernetes.io/region, failure-domain.kubernetes.io/zone, kubernetes.io/arch, kubernetes.io/hostname, kubernetes.io/instance-type, kubernetes.io/os)
```

**Anything else you would like to add:**

It should be great to add node-role in proposal in this list: https://github.com/kubernetes/enhancements/blob/master/keps/sig-auth/0000-20170814-bounding-self-labeling-kubelets.md#proposal

/kind feature

### PR DESCRIPTION
**Pull request by neeleshkorade**
**What this PR does / why we need it**:
Adds troubleshooting guide to the docs with an initial entry with a tip for the issue #1719

**Which issue(s) this PR fixes**
Fixes #1719

### CODE REVIEW

File: docs/book/src/user/troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Troubleshooting
+
+## Labeling nodes with reserved labels such as `node-role.kubernetes.io` fails with `kubeadm` error during bootstrap.
+
+Reserved labels cannot be set during bootstrapping (for example, by using `node-labels` in `kubeletExtraArgs` when bootstrapping with [CAPBK](https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm)) due to a [restriction](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#noderestriction) on the kubelet self-assigning reserved labels such as `node-roles`.  
+
+In order to achieve this, assinging of such labels to nodes needs to be done after the bootstrap process is over. One way to do that is to introduce a separate step that does the job of applying reserved labels after the bootstrap is over using kubectl as shown below.
```
Comment (detiber) [inline docs/book/src/user/troubleshooting.md]:
```suggestion
In order to achieve this, assigning of such labels to nodes needs to be done after the bootstrap process is over. One way to do that is to introduce a separate step that does the job of applying reserved labels after the bootstrap is over using kubectl as shown below.
```

Comment (neeleshkorade) [reply docs/book/src/user/troubleshooting.md]: Thanks @detiber.  I updated the PR

File: docs/book/src/user/troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Troubleshooting
+
+## Labeling nodes with reserved labels such as `node-role.kubernetes.io` fails with `kubeadm` error during bootstrap.
+
+Reserved labels cannot be set during bootstrapping (for example, by using `node-labels` in `kubeletExtraArgs` when bootstrapping with [CAPBK](https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm)) due to a [restriction](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#noderestriction) on the kubelet self-assigning reserved labels such as `node-roles`.
```
Comment (neolit123) [inline docs/book/src/user/troubleshooting.md]: > by using `node-labels`

should this have been:
> by using `node-roles` labels

?

Comment (neolit123) [reply docs/book/src/user/troubleshooting.md]: > self-assigning reserved labels such as `node-roles`. 

-> `node-role` (singular)

Comment (ncdc) [reply docs/book/src/user/troubleshooting.md]: @neolit123 please file an issue and/or if you have time, create a follow-up PR. This has already merged. Thanks!

**Pull request merged**